src/org/pentaho/metadata/registry/OrderedFileRegistry.java

```diff
@@ -126,9 +125,8 @@ protected void addToBuffer( Link link, StringBuffer buffer, int idx ) {
   }
 
   protected void addToBuffer( String id, String value, StringBuffer buffer ) {
-    Encoder encoder = ESAPI.encoder();
-    buffer.append( "<entry key=\"" ).append( encoder.encodeForXMLAttribute( id ) ).append( "\">" ).append(
-        encoder.encodeForXML( value ) ).append( "</entry>\n" );
+    buffer.append( "<entry key=\"" ).append( Encode.forXml( id ) ).append( "\">" ).append(
```

Comment:
I believe you need forXmlAttribute for this. The rules for an XML element are very different than the rules for an XML attribute (e.g. definitely not &, <, and not the string used to open the attribute)